utils/upload_ansible_roles_to_galaxy.py

```diff
@@ -362,10 +445,20 @@ def locally_clone_and_init_repositories(organization, repo_list):
 def main():
     args = parse_args()
 
-    all_role_whitelist = {"rhel7-role-%s" % p for p in PROFILE_WHITELIST}
+    all_product_whitelist = {"%s" % p for p in PRODUCT_WHITELIST}
+    product_whitelist = set(all_product_whitelist)
+    if args.product:
+        selected_products = {"%s" % p for p in args.product}
+        product_whitelist.intersection_update(selected_products)
+
+    all_role_whitelist = []
+    for product in product_whitelist:
+        for p in PROFILE_WHITELIST:
+            all_role_whitelist.append("%s-role-%s" % (product, p))
+
     role_whitelist = set(all_role_whitelist)
     if args.profile:
-        selected_roles = {"rhel7-role-%s" % p for p in args.profile}
+        selected_roles = {"%s" % p for p in args.profile}
```

Comment:
`selected_roles = set(args.profile)` should do the same job, but it's more understandable.

Reply:
Let's assume I run the script with `--product rhel8 --profile ospp`. At this point, the variables contain:
```
role_whitelist = {'rhel8-role-nist-800-171-cui', 'rhel8-role-pci-dss', 'rhel8-role-ospp', 'rhel8-role-stig-rhel7-disa', 'rhel8-role-hipaa', 'rhel8-role-C2S', 'rhel8-role-cjis', 'rhel8-role-rht-ccp'}
selected_roles = {'ospp'} 
```
The intersection of those 2 sets is obviously an empty set, so after executing the line below the `role_whitelist` is an empty set. That means no role will be updated if user specifies `--profile`.

Reply:
@jan-cerny this should be working now.